extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -31,20 +31,22 @@ import {
   getCorrelator,
 } from '../../../ads/google/a4a/utils';
 import {getLifecycleReporter} from '../../../ads/google/a4a/performance';
-import {documentStateFor} from '../../../src/document-state';
 import {getMode} from '../../../src/mode';
 import {stringHash32} from '../../../src/crypto';
 import {domFingerprintPlain} from '../../../src/utils/dom-fingerprint';
+import {viewerForDoc} from '../../../src/viewer';
 
 /** @const {string} */
 const ADSENSE_BASE_URL = 'https://googleads.g.doubleclick.net/pagead/ads';
 
-/** @const {!Object<string, string>} */
+/**
+ * See `VisibilityState` enum.
+ * @const {!Object<string, string>}
+ */
 const visibilityStateCodes = {
```

Comment:
[Older IE](https://msdn.microsoft.com/en-us/library/windows/apps/hh453385.aspx) gave "preview" as a visibilityState. A quick search of AdSense logs show no uses of &vis=4, so it can safely be removed. 

Current Google AdSense code does not (yet) know about "unloaded" state, so there is no value for the "vis" parameter corresponding to that state. I suggest we add that, with a value of 5.